src/low-level/imap/acl.c

```diff
@@ -347,6 +347,7 @@ int mailimap_acl_myrights(mailimap * session,
           * result = (struct mailimap_acl_myrights_data *)ext_data->ext_data;
           /* remove the element from rsp_extension_list */
           clist_delete(session->imap_response_info->rsp_extension_list, cur);
+          free(ext_data);
```

Comment:
can you use `mailimap_extension_data_free(ext_data)`.

You'll need to change the implementation of mailimap_extension_data_free  to:

```
mailimap_extension_data_free(struct
        mailimap_extension_data * data)
{
  if (data == NULL)
    return;

  if (data->ext_extension != NULL)
    data->ext_extension->ext_free(data);
  free(data);
}
```

The initial implementation seemed broken.

Reply:
Sure, but I see `* result = (struct mailimap_acl_myrights_data *)ext_data->ext_data;`
I think the innerData would be passed to the other one
Is it safe to use mailimap_extension_data_free ?

Reply:
Yes. if you use `ext_data->ext_data = NULL`.